packages/fe-fpm-writer/src/common/validate.ts

```diff
@@ -58,3 +59,29 @@ export function validateBasePath(basePath: string, fs?: Editor, dependencies = [
 export function getErrorMessage(error: Error | unknown): string {
     return error instanceof Error ? error.message : String(error);
 }
+
+/**
+ * Validates base path, checks and adds missing libraries.
+ *
+ * @param {string} basePath - the base path
+ * @param {Editor} fs - the memfs editor instance
+ * @param {boolean} addMissingLibraries - indicates whether to add missing libraries or throw an error
+ */
+export function checkRequiredLibraries(basePath: string, fs: Editor, addMissingLibraries = false): void {
```

Comment:
"check" should just check and return a result
"add" should add
and maybe there could be a method "ensureNeededDependencies" or similar, that first calls check, and based on the result calls add.

Reply:
Reviewed by Andis https://github.com/SAP/open-ux-tools/pull/2284#pullrequestreview-2265845394